### DISCUSSION TITLE
How to contribute to a translation?

### DISCUSSION THREAD

**Discussion by imflash217** [SelfControlApp/selfcontrol#739] [Q&A] [answered]
Hi,
Great app!!
How can I contribute to do a translation. I am not able to find any details about it in README or elsewhere.
Thanks.

**Comment by cstigler** [comment, marked as answer]
Hi @imflash217 , thanks so much for offering help with the translations! It does require a bit of technical knowledge to do that currently, but nothing too tricky.

If you're contributing additions/changes to an _existing_ translation, the easiest thing is probably to go to the `[LANG].lproj` folder (e.g. [sv.lproj](https://github.com/SelfControlApp/selfcontrol/tree/master/sv.lproj) for Swedish) and make changes to the `.strings` files there.

If you'd like to make a _new_ translation (!), probably the easiest thing to do is to duplicate one of those `.lproj` folders and rename it to the language code you're translating into (e.g. `es.lproj` for Spanish), and then edit the `.strings` files to put in translations for each of the strings. If you've got the setup to do Mac development, you can build and test your new translation yourself, but otherwise just open a PR and pass it to me, and I'm happy to do the testing for you.

Let me know if that makes sense!
